### DISCUSSION TITLE
How to change color of the focused list item (title and description)?

### DISCUSSION THREAD

**Discussion by koddr** [charmbracelet/bubbles#111] [Q&A] [answered]
Hi,

Can you tell me, how I can change the color of a list item (title and description) that is _currently_ in focus? 
I can't find it in the source code. 

![Screenshot 2022-01-30 at 15 28 12](https://user-images.githubusercontent.com/11155743/151699714-51d0d9fb-3974-41b1-9ce5-94bd7bebb8a8.png)

Can you please give me a hint? 😉

**Comment by zmnpl** [comment]
Funny enough I just came here too see if I’m missing something/ someone else has asked this. First issue up top it is 😆 Within the styles which can be set on a list I cannot see one that applies to the list items…

**Comment by meowgorithm** [comment, marked as answer]
Hi! Rendering (and behavior) for `list` items is done via the [`ItemDelegate`](https://pkg.go.dev/github.com/charmbracelet/bubbles@v0.10.2/list#ItemDelegate) interface. It can be a little confusing at first, but it allows the list to be very flexible and powerful.

If you just want to alter the default style you could do something like:

```go
import "github.com/charmbracelet/bubbles/list"

// Create a new default delegate
d := list.NewDefaultDelegate()

// Change colors
c := lipgloss.Color("#6f03fc")
d.Styles.SelectedTitle = d.Styles.SelectedTitle.Foreground(c).BorderLeftForeground(c)
d.Styles.SelectedDesc = d.Styles.SelectedTitle.Copy() // reuse the title style here

// Initailize the list model with our delegate
width, height := 80, 40
l := list.New(listItems, d, width, height)

// You can also change the delegate on the fly
l.SetDelegate(d)
```

This code would replace [this line](https://github.com/charmbracelet/bubbletea/blob/master/examples/list-default/main.go#L77) in the [`list-default` example](https://github.com/charmbracelet/bubbletea/tree/master/examples/list-default).

For full control over the way list items are rendered you can also define your own `ItemDelegate` too ([example](https://github.com/charmbracelet/bubbletea/blob/a6f46172ec4436991b90c2270253b2d212de7ef3/examples/list-simple/main.go#L28-L49)).

**Comment by zmnpl** [comment]
Ahh alright, thank you!
For me this is answered. list/defaultitem.go gives more insight (now that this understood :) ) about how the defaults styles are set up etc. 

Cheers!

**Comment by koddr** [comment]
OMG! Well, of course this is not done in the `list` style, but in the `item` style settings! 😞 It might be better to look in the source code, but if there is a way to specify it in the README (for everyone), that would be great.

Thanks, @meowgorithm

**Reply by meowgorithm** [reply]
Good call. The docs here are still very lacking and it’s something in our to-do list!
